query/src/sql/exec/mod.rs

```diff
@@ -12,124 +12,109 @@
+
+    pub fn spawn(mut self) -> Result<NewPipeline> {
+        let expr = self.expression.clone();
+        self.build_pipeline(&expr)?;
+        Ok(self.pipeline)
     }
 
-    #[async_recursion(? Send)]
-    pub async fn build_pipeline(&self, expression: &SExpr) -> Result<Pipeline> {
+    fn build_pipeline(&mut self, expression: &SExpr) -> Result<DataSchemaRef> {
         if !check_physical(expression) {
-            return Err(ErrorCode::LogicalError(format!(
-                "Invalid physical plan: {:?}",
-                expression
-            )));
+            return Err(ErrorCode::LogicalError("Invalid physical plan"));
         }
 
-        match expression.plan().as_ref() {
-            Plan::PhysicalScan(scan) => self.build_table_scan(scan).await,
-            Plan::PhysicalProject(project) => {
-                self.build_project(project, expression.children().as_slice())
-                    .await
+        let plan = expression.plan().clone();
+
+        match plan.plan_type() {
+            PlanType::PhysicalScan => {
+                let physical_scan = plan.as_any().downcast_ref::<PhysicalScan>().unwrap();
+                self.build_physical_scan(physical_scan)
             }
-            _ => Err(ErrorCode::LogicalError(format!(
-                "Invalid physical plan: {:?}",
-                expression
-            ))),
+            PlanType::Project => {
+                let project = plan.as_any().downcast_ref::<ProjectPlan>().unwrap();
+                self.build_project(project, &expression.children()[0])
+            }
+            _ => Err(ErrorCode::LogicalError("Invalid physical plan")),
         }
     }
 
-    async fn build_project(
-        &self,
-        project: &PhysicalProject,
-        children: &[SExpr],
-    ) -> Result<Pipeline> {
-        let child = &children[0];
-        let input_schema = Arc::new(DataSchemaHelper::build(child, &self.metadata)?);
-        let output_schema = Arc::new(DataSchemaHelper::build_project(
-            project,
-            child,
-            &self.metadata,
-        )?);
+    fn build_project(&mut self, project: &ProjectPlan, child: &SExpr) -> Result<DataSchemaRef> {
+        let input_schema = self.build_pipeline(child)?;
+        let schema_builder = DataSchemaBuilder::new(&self.metadata);
+        let output_schema = schema_builder.build_project(project, input_schema.clone())?;
 
-        let mut exprs = vec![];
-        for item in project.items.iter() {
-            let alias = self.metadata.column(item.index).name.clone();
-            let builder = ExpressionBuilder::create(&self.metadata);
-            let expr = builder.build(&item.expr)?;
-            let expr = Expression::Alias(alias, Box::new(expr));
-            exprs.push(expr);
+        let mut expressions = vec![];
+        let expr_builder = ExpressionBuilder::create(&self.metadata);
+        for expr in project.items.iter() {
+            let scalar = expr.expr.as_any().downcast_ref::<Scalar>().unwrap();
```

Comment:
How about using `expect` so that we can know more about this panic?

Reply:
I was considering introducing a `assert_downcast` function later.